packages/flutter_tools/lib/src/commands/downgrade.dart

```diff
@@ -191,11 +203,19 @@ class DowngradeCommand extends FlutterCommand {
       if (sha == null) {
         continue;
       }
-      final RunResult parseResult = await _processUtils!.run(<String>[
-        'git', 'describe', '--tags', sha,
-      ], workingDirectory: workingDirectory);
+      final RunResult parseResult = await _processUtils!.run(
+        <String>['git', 'describe', '--tags', sha],
+        workingDirectory: workingDirectory,
+      );
       if (parseResult.exitCode == 0) {
-        buffer.writeln('Channel "${getNameForChannel(channel)}" was previously on: ${parseResult.stdout}.');
+        if (buffer.isEmpty) {
+          buffer.writeln();
+        }
```

Comment:
Why is this extra `writeln` call necessary? Is the condition meant to be negated here (I don't know this code).

Reply:
This adds an extra line between the help text and the list of channel last versions:

```
To switch to a specific Flutter version, see: https://flutter.dev/to/switch-flutter-version

Channel "master" was previously on: v1.2.3.
Channel "beta" was previously on: v1.0.0.
```

Without this, the output would be:

```
To switch to a specific Flutter version, see: https://flutter.dev/to/switch-flutter-version
Channel "master" was previously on: v1.2.3.
Channel "beta" was previously on: v1.0.0.
```

Reply:
ah. understood—thanks!